src/wipplpy/modules/connection.py

```diff
@@ -0,0 +1,118 @@
+        """
+        Determine whether a local or remote connection is necessary, and then
+        call the corresponding method.
+
+        Parameters
+        ----------
+        shot_number : `int`
+            The shot number from which to extract MDSplus data.
+        tree_name : `str`
+            String representing the tree name of the device's MDSplus database.
+        server_name : `str`
+            String representing the server in which the shot's data is located.
+        """
+        local_server = socket.gethostname()
+        if local_server == server_name:
+            try:
+                self._local_connect(shot_number, tree_name)
+            except Exception as e:
+                # Local connection failed. Try a remote conn. for completeness
+                logging.debug(
+                    "Failed attempted local connection: `%s`\nTrying a"
+                    " remote connection.",
+                    e,
+                )
+                try:
+                    self._remote_connect(shot_number, tree_name, server_name)
+                except Exception as e:
+                    # Neither connection type succeeded. Raise the exception
+                    self.is_remote_connection = None  # reset the flag
+                    logging.exception(
+                        "An error occurred while attempting to connect to"
+                        " `%s`: `%s`",
+                        server_name,
+                        e,
+                    )
+                    raise
+
+        else:
+            try:
+                self._remote_connect(shot_number, tree_name, server_name)
+            except Exception as e:
+                # Remote connection failed. Try a local conn. for completeness
+                logging.debug(
+                    "Failed attempted remote connection: `%s`\nTrying a"
+                    " local connection.",
+                    e,
+                )
+                try:
+                    self._local_connect(shot_number, tree_name)
+                except Exception as e:
+                    # Neither connection type succeeded. Raise the exception
+                    self.is_remote_connection = None  # reset the flag
+                    logging.exception(
+                        "An error occurred while attempting to connect to"
+                        " `%s`: `%s`",
+                        server_name,
+                        e,
+                    )
+                    raise
+
+    def _local_connect(self, shot_number, tree_name):
+        """
+        Open a local MDSplus tree database.
+
+        Parameters
+        ----------
+        shot_number : `int`
+            The shot number from which to extract MDSplus data.
+        tree_name : `str`
+            String representing the tree name of the device's MDSplus database.
+        """
+        self.is_remote_connection = False
+
+        self.tree = mds.Tree(tree_name, shot_number)
```

Comment:
Just above this line you should have a logging statement of "trying to make local connection to ....".

Reply:
I will need to pass `server_name` as an argument to `_local_connect` to do this (unless you see a better way of doing so). Notably, the server name isn't needed for local connections, but it could still be useful (and not harmful) to have a dynamic reference to it (as an argument).

Reply:
Maybe then you can just have it say "Trying to make local connection to shot # ... on tree ..."?